src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -793,10 +794,19 @@ private String initializeDeviceId() {
         invalidIds.add("Android");
         invalidIds.add("DEFACE");
 
+        // see if device id already stored in db
+        DatabaseHelper dbHelper = DatabaseHelper.getDatabaseHelper(context);
+        String deviceId = dbHelper.getValue(DEVICE_ID_KEY);
+        if (!TextUtils.isEmpty(deviceId) || invalidIds.contains(deviceId)) {
+            return deviceId;
+        }
+
+        // see if device id stored in old sharedPrefs
```

Comment:
I think there should be an upgrade path that happens where upgradePrefs happens, and the getDeviceId code should only look in the "current version" store